File: 1-introduccion-a-la-seguridad-informatica/confidencialidad/Cesar.java
public class Cesar {

	public static String alfabeto = "abcdefghijklmnopqrstuvwxyz ";
	public static int movimiento = 3;
	public static void main(String args[]) {
		String mensajeEnClaro = "desarrollo seguro de plataformas web";

		String mensajeCifrado = "";
		for (int i = 0; i < mensajeEnClaro.length(); i++) {
			char c = mensajeEnClaro.charAt(i);
			char caracterCifrado = getCaracterCifrado(c);
			if (caracterCifrado == '\u0000') {
				System.out.println("Error: Se ingresó un caracter que no se pudo cifrar.");
				return;
			} else {
				mensajeCifrado += caracterCifrado;
			}
		}
		System.out.println(mensajeCifrado);
	}

	public static char getCaracterCifrado(int caracterSinCifrar) {
		for (int i = 0; i < alfabeto.length(); i++) {
			char c = alfabeto.charAt(i);
			if (caracterSinCifrar == alfabeto.charAt(i)) {
				return alfabeto.charAt((i + movimiento) % alfabeto.length());
			}
		}
		return '\u0000';
	}
}

File: 1-introduccion-a-la-seguridad-informatica/confidencialidad/cesar.py
class Cesar:
    ALFABETO = "abcdefghijklmnñopqrstuvwxyz "

    def __init__(self, movimiento=3):
        self.movimiento = movimiento

    def cifrar(self, mensaje):
        mensaje_cifrado = ""
        for car in mensaje:
            mensaje_cifrado += self._get_caracter_cifrado(car)

        return mensaje_cifrado

    def _get_caracter_cifrado(self, car_no_cifrado):
        i_nc = self.ALFABETO.find(car_no_cifrado)  # Índice del caracter dentro del alfabeto.

        if i_nc < 0:
            raise Exception(f"Caracter inválido: {car_no_cifrado}")

        i_cf = (i_nc + self.movimiento) % len(self.ALFABETO)  # Nuevo índice

        caracter_cifrado = self.ALFABETO[i_cf]

        return caracter_cifrado


if __name__ == "__main__":
    mensaje = "desarrollo seguro de plataformas web"
    cifrador = Cesar()
    mensaje_cifrado = cifrador.cifrar(mensaje)

    print(mensaje_cifrado)
